runtime/commonMain/src/kotlinx/serialization/internal/Tuples.kt

```diff
@@ -56,38 +57,44 @@ public sealed class KeyValueSerializer<K, V, R>(
     }
 }
 
+// todo: move from internal package and add documentation
```

Comment:
Could you please create a corresponding issue for each TODO?
It would be much easier to find a quick/small pieces of work, chances that we forget something are lower and reviews are quicker

Reply:
I've created a task list here: https://github.com/Kotlin/kotlinx.serialization/issues/298#issuecomment-577272487